Doc/library/typing.rst

```diff
@@ -572,6 +572,34 @@ These can be used as types in annotations and do not support ``[]``.
    * Every type is compatible with :data:`Any`.
    * :data:`Any` is compatible with every type.
 
+.. data:: Never
+
+   The `bottom type <https://en.wikipedia.org/wiki/Bottom_type>`_,
+   a type that has no members.
+
+   This can be used to define a function that should never be
+   called, or a function that never returns::
```

Comment:
I would scrap the "or a function that never returns" part here, if we want people to carry on using the `NoReturn` name for that purpose. Let's be really clear about the intended difference between the two.

Reply:
No, we should not encourage users to use `NoReturn`. We should just not kill it. I think `-> Never` is just as clear as `-> NoReturn` -- possibly clearer.

Reply:
Perhaps it's the `versionadded` section below that's making me uncomfortable — the "on older Python versions" wording makes it sound (to me) like it's saying "using `NoReturn` to do any of the above things is now discouraged ". Which sounds a lot like deprecation — but we're not deprecating.

I agree that `-> Never` is just as clear.

Reply:
No, discouraging is not the same as deprecating. (E.g. PEP 8 discourages lots of things that are not and will never be deprecated.) I think we *should* nudge people in the direction of `Never` (TOOWTDI and all that), but we needn't deprecate `NoReturn` until 3.10 is a distant memory.

Reply:
Okay.